source/wrapper/contracts/Wrapper.sol

```diff
@@ -74,42 +75,65 @@ contract Wrapper {
     require(order.signature.v != 0,
       "SIGNATURE_MUST_BE_SENT");
 
-    // The sender is sending ether that must be wrapped.
-    if (order.sender.token == address(wethContract)) {
+    // Check if sender is sending ether that must be wrapped.
+    _wrapEther(order.sender);
 
-      // Ensure message value is sender param.
-      require(order.sender.param == msg.value,
+    // Perform the swap.
+    swapContract.swap(order);
+
+    // Check if sender is receiving ether that must be unwrapped.
+    _unwrapEther(order.sender.wallet, order.signer.token, order.signer.param);
+  }
+
+  function provideDelegateOrder(
+    Types.Order calldata order,
+    IDelegate delegate
+  ) external payable {
+    // Delegate trade wallet must be order sender - checked in Delegate
+    // Signature must be sent - checked in Delegate
+
+    // Check if signer is sending ether that must be wrapped.
```

Comment:
Change to // Perform wrapping ether action if signer sends ether.
to indicate that this function is not just a check but also performs the state change